src/components/DecodeTxs/index.tsx

```diff
@@ -98,44 +98,35 @@ export const BasicTxInfo = ({
   )
 }
 
-export const getParameterElement = (parameter: DecodedDataBasicParameter, index: number): ReactElement => {
+export const getParameterElement = ({ name, type, value }: DecodedDataBasicParameter, index: number): ReactElement => {
   let valueElement
 
-  if (parameter.type === 'address') {
-    valueElement = (
-      <PrefixedEthHashInfo
-        hash={parameter.value}
-        showAvatar
-        textSize="lg"
-        showCopyBtn
-        explorerUrl={getExplorerInfo(parameter.value)}
-      />
-    )
-  }
-
-  if (parameter.type === 'bytes') {
-    valueElement = (
-      <FlexWrapper margin={5}>
-        <Text size="lg">{getByteLength(parameter.value)} bytes</Text>
-        <CopyToClipboardBtn textToCopy={parameter.value} />
-      </FlexWrapper>
-    )
+  if (!Array.isArray(value)) {
+    switch (type) {
+      case 'address':
+        valueElement = (
+          <PrefixedEthHashInfo hash={value} showAvatar textSize="lg" showCopyBtn explorerUrl={getExplorerInfo(value)} />
+        )
+        break
+      case 'bytes':
+        valueElement = (
+          <FlexWrapper margin={5}>
+            <Text size="lg">{getByteLength(value)} bytes</Text>
+            <CopyToClipboardBtn textToCopy={value} />
+          </FlexWrapper>
+        )
+        break
+    }
   }
 
   if (!valueElement) {
-    let value = parameter.value
-    if (parameter.type.endsWith('[]')) {
-      try {
-        value = JSON.stringify(parameter.value)
-      } catch (e) {}
-    }
-    valueElement = <Text size="lg">{value}</Text>
+    valueElement = <Text size="lg">{JSON.stringify(value)}</Text>
```

Comment:
stringifying a string?

Reply:
With what input? Objects with circular refs?

Reply:
In context, true. I'm just cautious of removing legacy try/catches.

Reply:
Yep, me and @iamacook also talked and agreed it won't throw in this situation. It would be more a matter of our convention in these situations but I'd opt to leave it out of the `try...catch`